core/src/main/scala/sttp/tapir/server/interceptor/reject/RejectHandler.scala

```diff
@@ -0,0 +1,46 @@
+package sttp.tapir.server.interceptor.reject
+
+import sttp.model.StatusCode
+import sttp.tapir._
+import sttp.tapir.server.interceptor.reject.RejectHandler._
+import sttp.tapir.server.interceptor.{RequestResult, ValuedEndpointOutput}
+
+trait RejectHandler {
+  def apply(failure: RequestResult.Failure): Option[ValuedEndpointOutput[_]]
+}
+
+object RejectHandler {
+
+  private[reject] def hasMethodMismatch(f: RequestResult.Failure): Boolean = f.failures.map(_.failingInput).exists {
+    case _: EndpointInput.FixedMethod[_] => true
+    case _                               => false
+  }
+
+  private[reject] object StatusCodeAndBody {
+    val NotFound: (StatusCode, String) = (StatusCode.NotFound, "Not Found")
+    val MethodNotAllowed: (StatusCode, String) = (StatusCode.MethodNotAllowed, "Method Not Allowed")
+  }
+}
+
+case class DefaultRejectHandler(response: (StatusCode, String) => ValuedEndpointOutput[_]) extends RejectHandler {
```

Comment:
maybe we could parametrise the `DefaultRejectHandler` with the optional status code to return if there's a non-fixed-method failure, and retain only one class (`DefaultRejectHandler`), with two pre-defined instances: `default` and `defaultOrNotFound`. 

I think it's easier to discover for people that they can choose an impl once they write `DefaultRejectHandler.(...)`

Reply:
Sounds good, I'm also not 100% happy with two different handler classes.